openslides_backend/action/actions/mediafile/calculate_mixins.py

```diff
@@ -57,3 +58,22 @@ def handle_children(
                         new_instance["is_public"],
                         new_instance["inherited_access_group_ids"],
                     )
+
+
+def calculate_inherited_groups_helper_with_parent_id(
+    datastore: DatastoreService,
+    access_group_ids: Optional[List[int]],
+    parent_id: Optional[int],
+) -> Tuple[bool, Optional[List[int]]]:
+    if parent_id:
+        parent = datastore.get(
+            FullQualifiedId(Collection("mediafile"), parent_id),
+            ["is_public", "inherited_access_group_ids"],
+        )
+        return calculate_inherited_groups_helper(
+            access_group_ids,
+            parent.get("is_public"),
+            parent.get("inherited_access_group_ids"),
+        )
+    else:
+        return (not bool(access_group_ids), access_group_ids)
```

Comment:
```suggestion
    if parent_id:
        parent = datastore.get(
            FullQualifiedId(Collection("mediafile"), parent_id),
            ["is_public", "inherited_access_group_ids"],
        )
    else:
        parent = {}
    return calculate_inherited_groups_helper(
        access_group_ids,
        parent.get("is_public"),
        parent.get("inherited_access_group_ids"),
    )
```
This was the initial hang up on why this change was needed, so the information about how to calculate these fields is gathered in one place ;)

Reply:
Done.